### PR TITLE
Open api corrector

### PR DESCRIPTION
**Pull request by StarProxima**

### CODE REVIEW

**Comment by StarProxima** [comment]
@Carapacik

File: swagger_parser/lib/src/parser/corrector/open_api_corrector.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import 'dart:convert' show jsonDecode;
+
+import 'package:yaml/yaml.dart';
+
+import '../../../swagger_parser.dart';
+import '../utils/case_utils.dart';
+
+/// Class used to correct data class, method and field names in OpenAPI
+class OpenApiCorrector {
+  OpenApiCorrector(this.config);
```
Comment (Carapacik) [inline swagger_parser/lib/src/parser/corrector/open_api_corrector.dart:10]: const and docs for constructor

**Review by Carapacik — CHANGES_REQUESTED** [review]

File: swagger_parser/lib/src/parser/corrector/open_api_corrector.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  /// [ParserConfig] that [OpenApiParser] use
+  final ParserConfig config;
+
+  Map<String, dynamic> correct() {
+    var fileContent = config.fileContent;
+
+    final definitionFileContent = config.isJson
+        ? jsonDecode(fileContent) as Map<String, dynamic>
```
[the remark this answers is not in the document]
Comment (StarProxima) [reply swagger_parser/lib/src/parser/corrector/open_api_corrector.dart:19]: okey, but.. why?
image

[the remark this answers is not in the document]
Comment (Carapacik) [reply swagger_parser/lib/src/parser/corrector/open_api_corrector.dart]: Why not?

**Review by Carapacik — APPROVED** [review]

File: swagger_parser/lib/src/parser/corrector/open_api_corrector.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (StarProxima) [reply swagger_parser/lib/src/parser/corrector/open_api_corrector.dart]: > Why not?

Well, if the Dart team, have made a wrapper over json.decode, then it's more readable and preferable. Well I've seen/written only jsonDecode/jsonEncode almost everywhere else

**Pull request merged**